AdyenNetworking/APIClient/APIClient.swift

```diff
@@ -172,34 +172,40 @@ public final class APIClient: APIClientProtocol {
     private func handle<R: Request>(
         _ result: URLSessionSuccess,
         _ request: R
-    ) throws -> HTTPResponse<R.ResponseType> {
+    ) throws -> HTTPDataResponse<R.ResponseType> {
         log(result: result, request: request)
+        
         do {
             if result.data.isEmpty, let emptyResponse = EmptyResponse() as? R.ResponseType {
-                return HTTPResponse(
+                return HTTPDataResponse(
                     headers: result.headers,
                     statusCode: result.statusCode,
-                    responseBody: emptyResponse
+                    responseBody: emptyResponse,
+                    responseData: result.data
```

Comment:
may i ask why do you need the `data`? its an empty response by definition, right?